frontend/src/app/event/event-editor/event-editor.component.ts

```diff
@@ -0,0 +1,140 @@
+import { Organization } from 'src/app/organization/organization.model';
+import { Event } from '../event.model';
+
+@Component({
+  selector: 'app-event-editor',
+  templateUrl: './event-editor.component.html',
+  styleUrls: ['./event-editor.component.css']
+})
+export class EventEditorComponent {
+  public static Route: Route = {
+    path: 'organizations/:slug/events/:id/edit',
+    component: EventEditorComponent,
+    title: 'Event Editor',
+    resolve: { profile: profileResolver, organization: organizationDetailResolver, event: eventDetailResolver }
+  };
+
+  /** Store the event to be edited or created */
+  public event: Event;
+  public organization_slug: string;
+  public organization: Organization;
+
+  public profile: Profile | null = null;
+
+  /** Stores whether the user has admin permission over the current organization. */
+  public adminPermission$: Observable<boolean>;
+
+  /** Add validators to the form */
+  name = new FormControl('', [Validators.required]);
+  time = new FormControl('', [Validators.required]);
+  location = new FormControl('', [Validators.required]);
+  description = new FormControl('', [Validators.required, Validators.maxLength(2000)]);
+  public = new FormControl('', [Validators.required]);
+
+  /** Create a form group */
+  public eventForm = this.formBuilder.group({
+    name: this.name,
+    time: new Date(this.time.value!),
+    location: this.location,
+    description: this.description,
+    public: (this.public.value! == "true")
+  });
+
+  constructor(private route: ActivatedRoute,
+    private router: Router,
+    protected formBuilder: FormBuilder,
+    protected organizationService: OrganizationService,
+    protected snackBar: MatSnackBar,
+    private eventService: EventService,
+    private permission: PermissionService) {
+
+    /** Get currently-logged-in user. */
+    const data = route.snapshot.data as { profile: Profile, organization: Organization, event: Event };
+    this.profile = data.profile;
+
+    /** Initialize event */
+    this.organization = data.organization;
+    this.event = data.event;
+    this.event.organization_id = this.organization.id;
+
+    /** Get ids from the url */
+    let organization_slug = this.route.snapshot.params['slug'];
+    this.organization_slug = organization_slug;
+
+    this.eventForm.setValue({
+      name: this.event.name,
+      time: this.event.time,
+      location: this.event.location,
+      description: this.event.description,
+      public: this.event.public
+    })
+
+    /** Set permission value */
+    this.adminPermission$ = this.permission.check('organization.events.manage', `organization/${this.organization!.id}`);
+  }
+
+  /** Event handler to handle submitting the Create Event Form.
+   * @returns {void}
+  */
+  onSubmit = () => {
+    if (this.eventForm.valid) {
+      Object.assign(this.event, this.eventForm.value)
+      if (this.event.id == -1) {

```

Comment:
This comment is based on the error I wrote about in the review. Here, `this.event` actually is being initialized to `null` by the `Object.assign(...)` call rather than `-1`. Changing either the conditional or the line above should solve this problem!

As a result, I think this condition always returning `false` is what was calling the `PUT` (update) API call rather than the `POST` (create) one.